include/boost/geometry/algorithms/detail/overlay/sort_by_side.hpp

```diff
@@ -316,7 +316,15 @@ public :
         // then take a point (or more) further back.
         // The limit of offset avoids theoretical infinite loops.
         // In practice it currently walks max 1 point back in all cases.
-        double const tolerance = 1.0e9;
+        // Use the coordinate type, but if it is too small (e.g. std::int16), use a double
+        using ct_type = typename geometry::select_most_precise
+            <
+                typename geometry::coordinate_type<Point>::type,
+                double
+            >::type;
```

Comment:
Required for types as `std::int16_t` to avoid overflow
(therefore it's tested in `union_other_types.cpp` now too)